libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/YamlConfigService.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.di.ipv.core.library.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import uk.gov.di.ipv.core.library.exceptions.ConfigParameterNotFoundException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.fasterxml.jackson.core.JsonParser.Feature.STRICT_DUPLICATE_DETECTION;
+import static java.util.Collections.emptyMap;
+
+public class YamlConfigService extends ConfigService {
+    private static final File PARAMETERS_FILE = new File("./core.local.params.yaml");
+    private static final File SECRETS_FILE = new File("./core.local.secrets.yaml");
+    private static final String PATH_SEPARATOR = "/";
+    private static final String CORE = "core";
+    private static final String FEATURE_SETS = "features";
+
+    private static final ObjectMapper YAML_OBJECT_MAPPER =
+            new ObjectMapper(new YAMLFactory()).configure(STRICT_DUPLICATE_DETECTION, true);
+
+    private final Map<String, String> parameters = new HashMap<>();
+    private final Map<String, Map<String, String>> featureSetParameters = new HashMap<>();
+    private final Map<String, String> secrets = new HashMap<>();
+
+    public YamlConfigService() {
+        this(PARAMETERS_FILE, SECRETS_FILE);
+    }
+
+    public YamlConfigService(File parametersFile, File secretsFile) {
+        try {
+            var paramsYaml = YAML_OBJECT_MAPPER.readTree(parametersFile).get(CORE);
+            var secretsYaml = YAML_OBJECT_MAPPER.readTree(secretsFile).get(CORE);
+
+            addConfig(parameters, paramsYaml);
+            addConfig(secrets, secretsYaml);
+
+            paramsYaml
```

Comment:
This'll have already been mapped into the `parameters` map I think - do we need to maintain a separate map? The override could just be looked up in the parameters map?

Reply:
Yeah, in my head it was to avoid iterating through the parameters, but I don't think we need to.